coll/adapt: Fix naming conventions and C11 atomic use

bsergentm opened the pull request:
This commit fixes some naming convention issues, such as function names
which should follow the naming ompi_coll_adapt instead of
mca_coll_adapt, reserved for component and module naming (cf. tuned
collective component);

It also fixes the use of _Atomic construct, which is only valid in C11.
OPAL constructs have already been adapted to that use, so use
opal_atomic_* types instead.

Signed-off-by: Marc Sergent <marc.sergent@atos.net>

Commit:
coll/adapt: Fix naming conventions and C11 atomic use

This commit fixes some naming convention issues, such as function names
which should follow the naming ompi_coll_adapt instead of
mca_coll_adapt, reserved for component and module naming (cf. tuned
collective component);

It also fixes the use of _Atomic construct, which is only valid in C11.
OPAL constructs have already been adapted to that use, so use
opal_atomic_* types instead.

Signed-off-by: Marc Sergent <marc.sergent@atos.net>

derbeyn commented on ompi/mca/coll/adapt/coll_adapt.h:
```diff
@@ -21,56 +21,56 @@
 
     /* MCA parameter: Priority of this component */
-    int adapt_priority;
+    int ompi_coll_adapt_priority;
```
the fields inside a structure do not have to follow the naming convention. You should have left them as they were

derbeyn replied:
You can refer to the component structure of coll/tuned, for example

bsergentm replied:
Indeed, done in 8b854f9fe3a23e39e3d3511808406b43bdf5c7d2

derbeyn commented on ompi/mca/coll/adapt/coll_adapt.h:
```diff
@@ -80,21 +80,21 @@ struct mca_coll_adapt_module_t {
 
     /* Whether this module has been lazily initialized or not yet */
-    bool enabled;
+    bool coll_adapt_enabled;
```
same as above for the module structure.

bsergentm replied:
Done in 8b854f9fe3a23e39e3d3511808406b43bdf5c7d2

derbeyn commented on ompi/mca/coll/adapt/coll_adapt.h:
```diff
@@ -80,21 +80,21 @@ struct mca_coll_adapt_module_t {
 
 /* Free ADAPT quest */
-int adapt_request_free(ompi_request_t ** request);
+int ompi_coll_adapt_request_free(ompi_request_t **request);
```
Since this routine is not used outside adapt, I think you can leave it as it was...
@bsergentm : can you please point me to the reference you're using for the naming conventions?

derbeyn replied:
I'm asking you to check if I'm wrong in my suggestions...

bsergentm replied:
It is indeed used only in adapt, but outside this file : in coll_adapt_ibcast.c and coll_adapt_ireduce.c.
As I understand, it should be considered as a public symbol.
The reference I'm using is https://github.com/open-mpi/ompi/wiki/CodingStyle, where they say that public symbols should be prefixed this way. Am I wrong here ?

FlorentGermain-Bull commented on ompi/mca/coll/adapt/coll_adapt_component.c:
```diff
@@ -53,19 +53,19 @@ mca_coll_adapt_component_t mca_coll_adapt_component = {
+	    ompi_coll_adapt_open,       /* open */
+	    ompi_coll_adapt_close,
+	    NULL,                       /* query */
+	    ompi_coll_adapt_register
```
```suggestion
	    .mca_open_component = ompi_coll_adapt_open,
	    .mca_close_component = ompi_coll_adapt_close,
	    .mca_register_component_params = ompi_coll_adapt_register
```
whith this, you keep in mind what you are setting and you do not need NULL lines

FlorentGermain-Bull replied:
you can apply this to l.50-53
see [coll_tuned_component.c](https://github.com/Bullrp4u/ompi/blob/758c3f8d05a476248db979a45fc16c961986285c/ompi/mca/coll/tuned/coll_tuned_component.c#L97-L109)
`coll_tuned_component.c

            .collm_version = {
            MCA_COLL_BASE_VERSION_2_0_0,

            /* Component name and version */
            .mca_component_name = "tuned",
            MCA_BASE_MAKE_VERSION(component, OMPI_MAJOR_VERSION, OMPI_MINOR_VERSION,
                                  OMPI_RELEASE_VERSION),


            /* Component open and close functions */
            .mca_open_component = tuned_open,
            .mca_close_component = tuned_close,
            .mca_register_component_params = tuned_register,
        },
`

bsergentm replied:
Done in 8b854f9fe3a23e39e3d3511808406b43bdf5c7d2

FlorentGermain-Bull commented on ompi/mca/coll/adapt/coll_adapt_module.c:
```diff
@@ -90,10 +90,10 @@ int mca_coll_adapt_init_query(bool enable_progress_threads, bool enable_mpi_thre
 						  int *priority)
 {
-    mca_coll_adapt_module_t *adapt_module;
+    mca_coll_adapt_module_t *coll_adapt_module;
```
I do not understand this renaming
As it is a local variable, is it necessary ?

bsergentm replied:
it is indeed not necessary.
Fixed in 8b854f9fe3a23e39e3d3511808406b43bdf5c7d2

FlorentGermain-Bull commented on ompi/mca/coll/adapt/coll_adapt.h:
```diff
@@ -21,56 +21,56 @@
+    opal_free_list_t *ompi_coll_adapt_ireduce_context_free_list;
+    opal_atomic_int32_t ompi_coll_adapt_ireduce_context_free_list_enabled;
 
 } mca_coll_adapt_component_t;
```
Do we realy need to rename these structure fields ?
We already know we are in adapt collective component because they are stored in the mca_coll_adapt_component global variable

bsergentm replied:
Indeed it is not useful.
Done in 8b854f9fe3a23e39e3d3511808406b43bdf5c7d2

FlorentGermain-Bull commented on ompi/mca/coll/adapt/coll_adapt.h:
```diff
@@ -80,21 +80,21 @@ struct mca_coll_adapt_module_t {
+
     /* Pointer to mca_coll_adapt_component */
-    mca_coll_adapt_component_t *adapt_component;
+    mca_coll_adapt_component_t *coll_adapt_component;
```
As the component is stored in the mca_coll_adapt_component global variable, is this pointer usefull ?

bsergentm replied:
It is indeed useless. Removed in af5f34efc3f84fb1bbac25a2cb4c251dc4802a74

FlorentGermain-Bull commented on ompi/mca/coll/adapt/coll_adapt_ireduce.c:
```diff
@@ -34,109 +34,109 @@ typedef int (*mca_coll_adapt_ireduce_fn_t) (const void *sbuf,
-    {4, (uintptr_t) mca_coll_adapt_ireduce_pipeline},
-    {5, (uintptr_t) mca_coll_adapt_ireduce_chain},
-    {6, (uintptr_t) mca_coll_adapt_ireduce_linear},
+static ompi_coll_adapt_algorithm_index_t ompi_coll_adapt_ireduce_algorithm_index[] = {
+    {0, (uintptr_t)ompi_coll_adapt_ireduce_tuned},
+    {1, (uintptr_t) ompi_coll_adapt_ireduce_binomial},
+    {2, (uintptr_t) ompi_coll_adapt_ireduce_in_order_binomial},
+    {3, (uintptr_t) ompi_coll_adapt_ireduce_binary},
+    {4, (uintptr_t) ompi_coll_adapt_ireduce_pipeline},
+    {5, (uintptr_t) ompi_coll_adapt_ireduce_chain},
+    {6, (uintptr_t) ompi_coll_adapt_ireduce_linear},
```
A type is defined for every collective function in ompi/mca/coll/coll.h:
`coll.h

    typedef int (*mca_coll_base_module_ireduce_fn_t)
      (const void *sbuf, void* rbuf, int count, struct ompi_datatype_t *dtype,
       struct ompi_op_t *op, int root, struct ompi_communicator_t *comm, ompi_request_t ** request,
       struct mca_coll_base_module_2_3_0_t *module);
`
Why using uintptr_t instead ?

FlorentGermain-Bull replied:
nvm, ompi_coll_adapt_algorithm_index_t uses a generic pointer

FlorentGermain-Bull reviewed (commented):
For me, component and module structure fileds renaming is a bit overkill

derbeyn commented:
sorry, When I saw that Florent had issues similar to mine, I realized I hadn't submitted my issues...

Commit:
coll/adapt: Remove unused component field in module

This commit removes an unneeded field referencing the component in the
module of adapt, as it is already available through the
mca_coll_adapt_component global variable.

Signed-off-by: Marc Sergent <marc.sergent@atos.net>
